### PR TITLE
crimson/common: read conf file in async

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2020-08-08_13:50:13-rados-wip-kefu-testing-2020-08-08-1822-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2020-08-09_03:30:55-rados-wip-kefu-testing-2020-08-08-1822-distro-basic-smithi/

failure in upgrade test also exists in master

- https://pulpito.ceph.com/kchai-2020-08-09_05:31:11-rados-master-distro-basic-smithi/

File: src/crimson/common/config_proxy.cc [hunk 1]
```diff
@@ -3,6 +3,20 @@
 
 #include "config_proxy.h"
 
+#if __has_include(<filesystem>)
```
Comment (rzarzynski) [inline src/crimson/common/config_proxy.cc:6]: My impression is we'll need to abstract this somehow in the future.

File: src/crimson/common/config_proxy.cc [hunk 2]
```diff
@@ -48,14 +62,36 @@ void ConfigProxy::show_config(ceph::Formatter* f) const {
...
-                                      &std::cerr,
-                                      CODE_ENVIRONMENT_DAEMON);
+  auto conffile_paths =
+    get_config().get_conffile_paths(*values,
+                                    conf_files.empty() ? nullptr : conf_files.c_str(),
+                                    &std::cerr,
+                                    CODE_ENVIRONMENT_DAEMON);
+  return seastar::do_with(std::move(conffile_paths), [this] (auto& paths) {
+    return seastar::repeat([path=paths.begin(), e=paths.end(), this]() mutable {
+      if (path == e) {
```
Comment (rzarzynski) [inline src/crimson/common/config_proxy.cc:72]: `e` was a little bit confusing. How about `end` or `end_`?

File: src/crimson/common/config_proxy.cc (hunk 1 above)
Comment (tchaikov) [reply src/crimson/common/config_proxy.cc:6]: once we bump up the seastar to its C++17 only version, we will have the luxury dropping all these `#if` in crimson.

File: src/crimson/common/config_proxy.cc (hunk 2 above)
Comment (tchaikov) [reply src/crimson/common/config_proxy.cc:72]: i'd avoid using the name collision. as `std::end()` is also named `end`. `e` is captured at line 71, and is used at line 72. after it's never used afterwards. so i think it is not worthy having a longer name.

File: src/crimson/common/config_proxy.cc [hunk 3]
```diff
@@ -46,5 +46,18 @@ void ConfigProxy::show_config(ceph::Formatter* f) const {
   get_config().show_config(*values, f);
 }
 
+seastar::future<> ConfigProxy::parse_config_files(const std::string& conf_files)
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/crimson/common/config_proxy.cc:63]: fixed.

File: src/common/config.cc [hunk 4]
```diff
@@ -352,7 +352,6 @@ int md_config_t::parse_config_files(ConfigValues& values,
 				    std::ostream *warnings,
 				    int flags)
 {
-
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply src/common/config.cc:355]: fixed.

**Comment by tchaikov** [comment]
@rzarzynski fixed and repushed.

**Pull request merged**